app/models/tag.rb

```diff
@@ -140,4 +199,23 @@ def entities_by_relationship_count(entity_type, page = 1)
   def total_count(entity_type)
     entities.where(primary_ext: entity_type).count
   end
+
+  # [ Hash ] -> Hash
+  #  example: output:
+  # {
+  #  "Relationship" => { 12 => <Relationship>, 22 => <Relationship> }
+  #  "Entity" => { 123 => <Entity>}
+  #  "List" => { 987 => <List> }
+  # }
+  #
+  def active_record_lookup_for_recent_edits(edits)
```

Comment:
my attempt at analyzing performance considerations here: we don't mind touching disk to query each resource individually here because we already paginated and thus are only ever doing a constant amount of single lookups. is that about right?

Reply:
edits contains at most 20 items, and we want to retrieve ActiveRecord models for each item.

The function will actually do at most  __3__ queries: one for Relationship, one for Entity, and one for List. ``` Entity.find( [ array of entity ids ] ) ``` issues only one query.